### ISSUE TITLE
[Performance] Speed-vs-memory regression on Conv3D FusedConv as of version 1.23.2

### ISSUE THREAD

**Issue by ryanseghers** [microsoft/onnxruntime#28466]
### Describe the issue

The peak CPU virtual memory allocation of some of our models went from 8 GB to 71 GB with onnxruntime-gpu change from version 1.23.0 to 1.23.2. This large allocation continues to happen on the latest version tested 1.25.1. This happens on both Windows and Linux, using the CPU execution provider.

Here is the message when the allocation fails the large allocation: 
onnxruntime.capi.onnxruntime_pybind11_state.RuntimeException: [ONNXRuntimeError] : 6 : RUNTIME_EXCEPTION : Non-zero status code returned while running FusedConv node. Name:'Conv__4386' Status Message: /onnxruntime_src/onnxruntime/core/framework/bfc_arena.cc:358 void* onnxruntime::BFCArena::AllocateRawInternal(size_t, bool, onnxruntime::Stream*) Failed to allocate memory for requested buffer of size 76672189440

It is possible to reduce the peak virtual memory usage using session options. Setting enable_cpu_mem_arena to False drops that peak alloc to ~40 GB. Changing intra_op_num_threads to 1 reduces that large alloc to the nominal value (~8 GB), however at that value the processing time is about doubled vs version 1.23.0. As intra_op_num_threads is increased the speed increases but so does the mem alloc size. So just using that session option I am unable to get back to the same speed-mem operating point, I am only able to choose between slower or larger virtual memory usage vs version 1.23.0.

### To reproduce

Here is a repro:
```
    tf.keras.backend.set_image_data_format("channels_last")
    x = tf.keras.Input((32, 64, 64, 1))
    m = tf.keras.Model(x, tf.keras.layers.Conv3D(6, 7, padding="same")(x))
    onnx.save(tf2onnx.convert.from_keras(m)[0], "{os.path.abspath(ONNX_FILE)}")

    sess = ort.InferenceSession("{os.path.abspath(ONNX_FILE)}", providers=["CPUExecutionProvider"])
    inp = sess.get_inputs()[0]
    sess.run(None, {{inp.name: np.random.randn(12, *inp.shape[1:]).astype(np.float32)}})
```

And vm peak across different onnxruntime versions:
    Testing onnxruntime-gpu 1.23.0 ... VmPeak = 3.6 GB
    Testing onnxruntime-gpu 1.23.2 ... VmPeak = 7.7 GB
    Testing onnxruntime-gpu 1.25.1 ... VmPeak = 7.7 GB

### Urgency

We are pinned to a version older than 1.23.2 and that avoids this issue.

### Platform

Windows

### OS Version

Windows 11

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.23.2

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

Default CPU

### Model File

    tf.keras.backend.set_image_data_format("channels_last")
    x = tf.keras.Input((32, 64, 64, 1))
    m = tf.keras.Model(x, tf.keras.layers.Conv3D(6, 7, padding="same")(x))
    onnx.save(tf2onnx.convert.from_keras(m)[0], "{os.path.abspath(ONNX_FILE)}")

### Is this a quantized model?

Unknown

**Comment by hariharans29** [comment]
@ryanseghers - Thanks for bringing this to my attention. Can you please try with the fix in #28482 ? Since you are using the CPU EP and not the CUDA EP, you can try building just the CPU package from source (from the referenced branch). It is easier to build than CUDA.

Note: I just fixed the excessive allocation. The algorithm for batched and/or grouped convolution also changed in 1.23.2 — we now parallelize across batch × group, instead of only across the output spatial map as in prior versions. I expect that algorithm change to improve perf for your use case.

**Comment by ryanseghers** [comment]
@hariharans29 Thank you for the fast reply. The fix looks good. I built and tested on Windows x64 and the speed and virtual memory peak match the 1.23.0 baseline. I didn't test on Linux because it looks like very low odds of different behavior there.
